store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedGraphStorage.java

```diff
@@ -603,33 +489,65 @@ private boolean changeGraphId(final String graphId, final String newGraphId, fin
         return rtn;
     }
 
-    private GraphConfig cloneGraphConfigWithNewGraphId(final String newGraphId, final Graph graphToMove) {
-        return new GraphConfig.Builder()
-                .json(new GraphSerialisable.Builder().graph(graphToMove).build().getConfig())
-                .graphId(newGraphId)
+    private void updateCacheWithNewGraphId(final String newGraphId, final GraphSerialisable graphToMove, final FederatedAccess access) throws StorageException {
+        //rename graph
+        GraphSerialisable updatedGraphSerialisable = new GraphSerialisable.Builder(graphToMove)
+                .config(cloneGraphConfigWithNewGraphId(newGraphId, graphToMove))
                 .build();
+
+        try {
+            this.put(updatedGraphSerialisable, access);
+        } catch (final Exception e) {
+            //TODO FS recovery
```

Comment:
What is this TODO describing?

Reply:
To try to possibly recover from the error. There is overlap with this and other fedstore branches.